examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,193 @@
+Comparing Linear Bayesian Regressors
+====================================
 
-Fit regression model with Bayesian Ridge Regression.
+This example compares two different bayesian regressors:
 
-See :ref:`bayesian_ridge_regression` for more information on the regressor.
+ - a :ref:`automatic_relevance_determination`
+ - a :ref:`bayesian_ridge_regression`
 
-Compared to the OLS (ordinary least squares) estimator, the coefficient
-weights are slightly shifted toward zeros, which stabilises them.
+In the first part, we use an :ref:`ordinary_least_squares` (OLS) model as a
+baseline for comparing the models' coefficients with respect to the true
+coefficients. Thereafter, we show that the estimation of such models is done by
+iteratively maximizing the marginal log-likelihood of the observations.
 
-The histogram of the estimated weights is very peaked, as a sparsity-inducing
-prior is implied on the weights.
+In the last section we plot predictions and uncertainties for the ARD and the
+Bayesian Ridge regressions using a polynomial feature expansion to fit a
+non-linear relationship between `X` and `y`.
 
-The estimation of the model is done by iteratively maximizing the
-marginal log-likelihood of the observations.
+"""
 
-We also plot predictions and uncertainties for ARD
-for one dimensional regression using polynomial feature expansion.
-Note the uncertainty starts going up on the right side of the plot.
-This is because these test samples are outside of the range of the training
-samples.
+# %%
+# Models robustness to recover the ground truth weights
+# =====================================================
+#
+# Generate synthetic dataset
+# --------------------------
+#
+# We generate a dataset where `X` and `y` are linearly linked: 10 of the
+# features of `X` will be used to generate `y`. The other features are not
+# useful at predicting `y`. In addition, we built a system where n_samples ==
+# n_features. Such setting is challenging for an OLS model and leads to
+# arbitrary large weights. Having a prior on the weights and a penalty
+# alleviates the problem. Finally, gaussian noise is added.
+
+from sklearn.datasets import make_regression
+
+X, y, true_weights = make_regression(
+    n_samples=100,
+    n_features=100,
+    n_informative=10,
+    noise=8,
+    coef=True,
+    random_state=42,
+)
 
-"""
+# %%
+# Fit the regressors
+# ------------------
+#
+# We now fit both Bayesian models and the OLS to later compare the models'
+# coefficients.
+
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression().fit(X, y)
+brr = BayesianRidge(compute_score=True, n_iter=30).fit(X, y)
+ard = ARDRegression(compute_score=True, n_iter=30).fit(X, y)
+df = pd.DataFrame(
+    {
+        "LinearRegression": olr.coef_,
+        "BayesianRidge": brr.coef_,
+        "ARDRegression": ard.coef_,
+        "Weights of true generative process": true_weights,
```

Comment:
I would still move this line in the first position in the dict since they are the true weights.